### PR TITLE
[Elastic Agent] Add options to bootstrap Fleet Server with TLS

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Adds command line arguments to `enroll` and `install` so Fleet Server can be bootstrapped with TLS enabled. By default if no options are provided a self-signed certificate is generated and the Elastic Agent communicates to the local bootstrapped Fleet Server with a pinned CA.

## Why is it important?

Required so Fleet Server is bootstrapped securely in the default case and provide the ability for custom certificates to be used for bootstrapping Fleet Server in production.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- elastic/fleet-server#90

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go [hunk 1]
```diff
@@ -222,6 +231,53 @@ func (c *EnrollCmd) fleetServerBootstrap(ctx context.Context) error {
 	return nil
 }
 
+func (c *EnrollCmd) prepareFleetTLS() error {
+	host := c.options.FleetServerHost
+	if host == "" {
+		host = "localhost"
+	}
+	port := c.options.FleetServerPort
+	if port == 0 {
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go:240]: can this be negative?

**Review by michalpristas — APPROVED** [review]
change looks good overall, we're starting to have a lot of flags
tested for regression on mac/linux seems ok

File: x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go [hunk 2]
```diff
@@ -222,6 +231,53 @@ func (c *EnrollCmd) fleetServerBootstrap(ctx context.Context) error {
...
+		port = defaultFleetServerPort
+	}
+	if c.options.FleetServerCert != "" && c.options.FleetServerCertKey == "" {
+		return errors.New("certificate private key is required when certificate provided")
+	}
+	if c.options.FleetServerCertKey != "" && c.options.FleetServerCert == "" {
+		return errors.New("certificate is required when certificate private key is provided")
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go:247]: these are printed to stdout and not logged, so I think its best to just make them simple messages so its clear when using the CLI

File: x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go (hunk 1 above)
Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go:240]: no its an `uint16`

**Pull request merged**